modules/howtos/examples/transactions-example.php

```diff
@@ -0,0 +1,489 @@
+      $qr = $ctx->query(
+        $st,
+        TransactionQueryOptions::build()
+          ->positionalParameters([99.99, $hotelChain, $country]));
+          // ->scopeQualifier("`travel-sample`.`inventory`"));
+
+      print_r($qr->metaData()->metrics());
+      if ($qr->metaData()->metrics()["mutationCount"] != 2) {
+        // throw new \Exception("Mutation count not the expected amount.");
+        echo "WARN: Mutation count not the expected amount.\n";
+      }
+    }
+  );
+  // end::queryExamplesUpdate[]
+
+  // tag::queryExamplesComplex[]
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) use ($hotelChain, $country) { 
+      // Find all hotels of the chain
+      $qr = $ctx->query(
+        'SELECT reviews FROM `travel-sample`.inventory.hotel WHERE url LIKE $1 AND country = $2',
+        TransactionQueryOptions::build()
+          ->positionalParameters([$hotelChain, $country]));
+          // ->scopeQualifier("`travel-sample`.`inventory`"));
+
+      // This function (not provided here) will use a trained machine learning model to provide a
+      // suitable price based on recent customer reviews.
+      function priceFromRecentReviews(Couchbase\QueryResult $qr) {
+          // this would call a trained ML model to get the best price
+          return 99.98;
+      }
+      $updatedPrice = priceFromRecentReviews($qr);
+
+      // Set the price of all hotels in the chain
+      $ctx->query(
+        'UPDATE `travel-sample`.inventory.hotel SET price = $1 WHERE url LIKE $2 AND country = $3',
+        TransactionQueryOptions::build()
+          ->positionalParameters([$updatedPrice, $hotelChain, $country]));
+          // ->scopeQualifier("`travel-sample`.`inventory`"));
+    }
+  );
+  // end::queryExamplesComplex[]
+}
+
+function queryInsert($cluster) {
+  // tag::queryInsert[]
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) { 
+      $ctx->query("INSERT INTO `travel-sample`.inventory.airline VALUES ('doc-c', {'hello':'world'})"); // <1>
+      $st = "SELECT `default`.* FROM `travel-sample`.inventory.airline WHERE META().id = 'doc-c'"; // <2>
+      $qr = $ctx->query($st);
+  });
+  // end::queryInsert[]
+}
+
+function queryRYOW($cluster) {
+  // tag::queryRYOW[]
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) {
+      $qr = $ctx->query("UPDATE `travel-sample`.inventory.hotel SET price = 99.00 WHERE name LIKE \"Marriott%\"");
+      if ($qr->metaData()->metrics()["mutationCount"] != 2) {
+        // throw new \Exception("Mutation count not the expected amount.");
+        echo "WARN: Mutation count not the expected amount.\n";
+      }
+    }
+  );
+// end::queryRYOW[]
+}
+
+function queryOptions($cluster) {
+  // tag::queryOptions[]
+
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) {
+      $txQo = TransactionQueryOptions::build()
+        ->timeout(1000)
```

Comment:
TransactionQueryOptions shouldn't have timeout: see https://hackmd.io/cHIcXWZSQOi23qpJJAa8OQ#TransactionQueryOptions for supported options
(We automatically set the timeout to what's left of the transaction)

Reply:
OK, I've added `->readonly(false)` here, suggestions for a better/more compelling example welcome!

Reply:
Pinging @avsej for visibility on the API issues